### DISCUSSION TITLE
Is it possible to set a custom instance for infinite queries?

### DISCUSSION THREAD

**Discussion by nerdyman** [orval-labs/orval#790] [Q&A]
Hi 👋

First, thanks for the awesome work on Orval! 

I've been migrating a codebase to use Orval with React Query v4 but haven't been able to figure out if it's possible to create a custom infinite query instance or inject additional code into the generated infinite queries. 

All of the infinite queries in our API follow the same pattern so I'd like to have a custom infinite query which binds the `select` property on the query to remap and add some properties, while also having the return type reflect these changes. I checked the docs and issues but couldn't find a solution, is it possible?

Thanks!

**Comment by anymaniax** [comment]
Hello @nerdyman, using the options queryOptions should solve your problem. You have an example of config [here](https://github.com/anymaniax/orval/issues/243#issuecomment-1448999922) the doc still need to be enhanced for that case

**Reply by nerdyman** [reply]
Thanks! I'll give that a go
